src/porepy/models/units.py

```diff
@@ -123,3 +125,73 @@ def W(self) -> number:
     def degree(self) -> number:
         """Angle unit, derived from rad."""
         return self.rad * 180 / np.pi
+
+    @overload
+    def convert_units(
+        self, value: number, units: str, to_si: Optional[bool] = False
+    ) -> number: ...
+
+    @overload
```

Comment:
That code is not from me, I just moved it from `MaterialConstants` to `Units` according to #1247 .
Could you specify what is missing?

Reply:
Damn! I might be the culprit myself. Isn't the purpose for mypy to know that if the method is called with value being a number, it will return a number? And same for ndarray. By the way, I think the same could be more elegantly achieved using a typevar or something of the sort.

Reply:
Yes, I mean it's working! And yes that is possible, with much less code.
`T = TypeVar('T', pp.number, np.ndarray)` would do the trick.

Reply:
Removed the overload and introduced a local type var, to be either a number or an array.